packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
+  hubType: HubType;
+  maintainer: IHubMember; // item.owner with more user metadata
+  access: { // overrides item.access with more attributes. could flatten.
+    visibility: 'private' | 'org' | 'public'; // item.access
+    permission?: 'view' | 'edit' | 'admin'; // item.itemControl
+    groups?: Array<IGroup>; // item.sharing.groups via content/users/:username/items/:id
+  };
+
+  // Explicit data information since this is a common confusion + bug report
+  createdDate: Date; // formal metadata || new Date(item.created)
+  createdDateSource: string; // description of what was used for this attribute
+  publishedDate: Date; // formal metadata || new Date(item.created)
+  publishedDateSource: string; // description of what was used for this attribute
+  updatedDate: Date; // formal metadata || new Date(item.modified)
+  updatedDateSource: string; // description of what was used for this attribute
+  
+  contentUrl: string; // Link to the raw content. item.url in most (but not all) item types
+  thumbnailUrl: string; // Full URL. item.thumbnail with host + path 
+  
+  license: IHubLicense; // [Future] item.licenseInfo 
+  boundary: IHubGeography; // [Future] Inline, or boolean when stored at known location /resources/boundary.json
+  
+  // Additional metadata from custom/formal elements
+  metadata?: {
+      // Unique or additional formal metadata that will be displayed in sidebar
+  };
+
+  // Hub configuration metadata
+  actionLinks?: IActionLink[]; // item.properties.links
+  actionConfigs?: Object; // item.properties.actions - enable/disable standard actions like `createWebmap` or `createStorymap`
```

Comment:
`Object` will accept primitive & non-primitive types. It may be more appropriate to to use `object` so the compiler throws errors if we try to assign a `string`, `number`, etc

Reply:
@ajturner is the idea here to attach in functions? or is this more a hash of settings for well-known-actions?

Reply:
i hope it's the latter. If so, I propose `actionOptions`?

Reply:
`actionConfigs` is a hash of settings for links. 

for example
```js
actionConfigs: [
  { title: "Open GeoHive Viewer", url: "https://hosting.geohive.ie/custom-viewer?layer=4ef"},
]
```